### PR TITLE
NEW update button added to checkout process

### LINKED ISSUE #11: Cart should have an "update" button, controller should use the backend data
Rather than update the cart during the actual checkout process (which opens up security risks for people maliciously modifying cart contents) and also requires another set of validation for the cart data that gets posted (#9), we should add an "Update" button underneath the cart table on the checkout page so that the user can update the quantities etc without it being part of the checkout form post.

This would mean the checkout post would read the cart from the backend instead of the post data, and removes the need for an extra step of validation of that cart data - we can assume all cart data is validated by the cart controller when adding or updating items in it.

Checkout controller is then only responsible for saving the cart items and handling the user data submission.

### PR DESCRIPTION
**Pull request by fspringveldt**
Introduced a new summary method to cart which displays it's content either in edit/view mode depending on Cart::$update setting.
This change caused issue #32 and was raised accordingly. 
Fixes #11

### CODE REVIEW

**Comment by fspringveldt** [comment]
Codecov dropped. I'll address that pronto.

File: css/dms-cart.css [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+  margin-right: 15px;
 }
 
 .checkout-page-thumbnail .thumbnail {
   width: 32px;
   border: none;
   vertical-align: middle;
 }
+
+.Actions a.action:hover{
```
Comment (sachajudd) [inline css/dms-cart.css]: Spaces, please follow above! (Also for line 12)

**Review by sachajudd — CHANGES_REQUESTED** [review]
Could you please rebase?

File: code/classes/DMSDocumentCart.php [hunk 2]
```diff
@@ -247,4 +254,52 @@ public function saveSubmission(Form $form)
...
+
+    /**
+     * Sets the updating flag
+     *
+     * @param bool $updating
+     * @return DMSDocumentCart
+     */
+    public function setViewOnly($updating)
+    {
+        $this->viewOnly = (bool) $updating;
```
Comment (robbieaverill) [inline code/classes/DMSDocumentCart.php]: might make more sense to call this `$viewOnly`

File: code/classes/DMSDocumentCart.php [hunk 3]
```diff
@@ -247,4 +254,52 @@ public function saveSubmission(Form $form)
...
+    public function getLink($action)
+    {
+        if ($url = array_search(
+            'DMSDocumentCartController',
+            (array)Config::inst()->get('Director', 'rules')
+        )
+        ) {
+            return Controller::join_links($url, $action);
+        }
+    }
```
Comment (robbieaverill) [inline code/classes/DMSDocumentCart.php]: Could you rubber duck the logic this method to me?

File: tests/controllers/DMSDocumentCartControllerTest.php [hunk 4]
```diff
@@ -166,7 +167,7 @@ public function testCannotAddMoreThanSuggestedQuantityOfItem()
     {
         $document = $this->objFromFixture('DMSDocument', 'limited_supply');
         $result = $this->get('/documentcart/add/' . $document->ID . '?quantity=5&ajax=1');
-        $this->assertContains('You can\'t add 5 of this document', (string) $result->getBody());
+        $this->assertContains('You can\'t add 5 of', (string) $result->getBody());
```
Comment (robbieaverill) [inline tests/controllers/DMSDocumentCartControllerTest.php]: Maybe this assertion could also cover the new "title" in the message too

**Review by robbieaverill — CHANGES_REQUESTED** [review]

File: tests/controllers/DMSDocumentCartControllerTest.php [hunk 5]
```diff
@@ -179,4 +180,69 @@ public function testValidationErrorReturnedOnInvalidAdd()
...
+        );
+    }
+
+    /**
+     * Tests that the cart summary view is returned.
+     */
+    public function testView()
+    {
+        $result = $this->get('documentcart/view');
+        $this->assertInstanceOf('SS_HTTPResponse', $result);
```
Comment (robbieaverill) [inline tests/controllers/DMSDocumentCartControllerTest.php:264]: this assertion seems a bit pointless TBH

**Review by robbieaverill — APPROVED** [review]

**Review by sachajudd — APPROVED** [review]
A few minor UX issues but these will be addressed in a later commit. Nice @fspringveldt will merge after tests.

**Pull request merged**